packages/webapp/src/utils/helpers.js

```diff
@@ -419,6 +419,42 @@ const validateTextInput = (text, hrefId, fieldType = 'input', maxLength = null,
   return error.err ? error : null
 }
 
+const checkForDuplicate = (array, property, value, hrefId, errorMessage, excludeIndex) => {
+  const duplicate = array.some((item, index) => {
+    if (typeof item === 'object' && item !== null && property in item) {
+      return index !== excludeIndex && item[property] && item[property].toLowerCase() === value.toLowerCase()
+    } else {
+      return index !== excludeIndex && item.toLowerCase() === value.toLowerCase()
+    }
+  })
+  if (duplicate) {
+    return {
+      err: {
+        text: errorMessage,
+        href: hrefId
+      }
+    }
+  }
+  return null
+}
+
+const checkForDuplicateConcatenated = (array, properties, targetObject, hrefId, errorMessage, excludedIndex) => {
+  const targetValue = properties.map(prop => targetObject[prop].toLowerCase()).join(' ').trim()
+  const error = {}
+  const duplicate = array.some((item, index) => {
+    if (excludedIndex !== null && index === excludedIndex) return false
+    const itemValue = properties.map(prop => item[prop]?.toLowerCase()).join(' ').trim()
+    return itemValue === targetValue
+  })
+  if (duplicate) {
+    error.err = {
+      text: errorMessage,
+      href: hrefId
+    }
+    return error
+  }
+  return null
+}
```

Comment:
If we are going with this approach ( I am not sure yet) then we need tests here

Reply:
yes added the tests. didnt know there was a test file